rfcs/010-react-intl.md

```diff
@@ -0,0 +1,53 @@
+# RFC 010 - Best Practices with react-intl
+
+## Affected projects
+
+- Frontend
+
+## Motivation
+
+We're using react-intl and have currently 2 ways of expressing a message ("defineMessages + formatMessage", or "FormattedMessage"). It's currently not clear when to use one or another approach and guidelines would be useful for current and new developers.
+
+## Comparaison of the 2 approaches
+
+### defineMessages + formatMessage
+
+Pros:
+- Without JSX "bloat", concise
+- Can be assigned to a variable
+- Can be post manipulated in Javascript
+- Support values
+
+Cons:
+- 2 steps: Needs first to defineMessages then formatMessage
```

Comment:
Stumbled upon this a few days ago, and it's actually a concrete problem as we're approaching our limit for number of strings on Crowdin.

```suggestion
- 2 steps: Needs first to defineMessages then formatMessage
- Unused strings are not automatically detected nor removed from the translations files, you need to remove them from the `defineMessages` first, and there's no linter warning to tell you wether they're used or not
```